scripts/calculate_percentile_offline.py

```diff
@@ -20,89 +20,123 @@
     daily_weather_data.drop(indexNames, inplace=True)
     # remove data recorded after END_YEAR
-    indexNames = daily_weather_data[daily_weather_data['year'] > END_YEAR].index
+    indexNames = daily_weather_data[daily_weather_data['year']
+                                    > END_YEAR].index
     daily_weather_data.drop(indexNames, inplace=True)
-    return 
+    return
+
 
 def remove_data_outside_fire_season():
+    print('remove_data_outside_fire_season...')
     # remove data recorded outside of fire season
-    indexNames = daily_weather_data[daily_weather_data['month'] < FIRE_SEASON_START_MONTH].index
+    indexNames = daily_weather_data[daily_weather_data['month']
+                                    < FIRE_SEASON_START_MONTH].index
     daily_weather_data.drop(indexNames, inplace=True)
-    indexNames = daily_weather_data[daily_weather_data['month'] > FIRE_SEASON_END_MONTH].index
+    indexNames = daily_weather_data[daily_weather_data['month']
+                                    > FIRE_SEASON_END_MONTH].index
     daily_weather_data.drop(indexNames, inplace=True)
-    indexNames = daily_weather_data[(daily_weather_data['month'] == FIRE_SEASON_START_MONTH) & (daily_weather_data['day'] < FIRE_SEASON_START_DATE)].index
+    indexNames = daily_weather_data[(daily_weather_data['month'] == FIRE_SEASON_START_MONTH) & (
+        daily_weather_data['day'] < FIRE_SEASON_START_DATE)].index
     daily_weather_data.drop(indexNames, inplace=True)
-    indexNames = daily_weather_data[(daily_weather_data['month'] == FIRE_SEASON_END_MONTH) & (daily_weather_data['day'] > FIRE_SEASON_END_DATE)].index
+    indexNames = daily_weather_data[(daily_weather_data['month'] == FIRE_SEASON_END_MONTH) & (
+        daily_weather_data['day'] > FIRE_SEASON_END_DATE)].index
     daily_weather_data.drop(indexNames, inplace=True)
     return
 
+
 def sort_by_weather_station():
+    print('sort_by_weather_station...')
     # sort the dataframe by station ID
     daily_weather_data.sort_values(by=['station_code'], inplace=True)
-    return 
+    return
+
 
 def calculate_percentile_per_station():
-    global ffmc_percentiles, bui_percentiles, isi_percentiles
-    ffmc_percentiles = daily_weather_data.groupby('station_code').ffmc.quantile(PERCENTILE)
-    bui_percentiles = daily_weather_data.groupby('station_code').bui.quantile(PERCENTILE)
-    isi_percentiles = daily_weather_data.groupby('station_code').isi.quantile(PERCENTILE)
+    print('calculate_percentile_per_station...')
+    global ffmc_percentiles, bui_percentiles, isi_percentiles, data_years
+    ffmc_percentiles = daily_weather_data.groupby(
+        'station_code').ffmc.quantile(PERCENTILE)
+    bui_percentiles = daily_weather_data.groupby(
+        'station_code').bui.quantile(PERCENTILE)
+    isi_percentiles = daily_weather_data.groupby(
+        'station_code').isi.quantile(PERCENTILE)
     return
 
+
+def list_years_per_station():
+    print('list_years_per_station...')
+    global data_years
```

Comment:
Can we give type hints on these global variables?
global data_years # Dict[int, List[int]]
like this? (it may not be technically a dictionary but it gives you the idea that it's a key, value pair)

Reply:
I've added some comments - but not given it an exact type hint, since that would be a lie. For example - it's not an int, it's int64, and it's not a dictionary, it's a pandas data series.